utils/watchcat/files/watchcat.init

```diff
@@ -90,53 +90,36 @@ load_watchcat() {
 		logger -p user.err -t "watchcat" "reboot program $1 not started - $error"
 		return
 	}
+	
+	procd_open_instance
 
+	# Need to conditionally run mode functions because they have different signatures
 	case "$mode" in
 	periodic_reboot)
-		/usr/bin/watchcat.sh "periodic_reboot" "$period" "$forcedelay" &
-		logger -p user.info -t "watchcat" "started task (mode=$mode;period=$period;forcedelay=$forcedelay)"
+		procd_set_param command /usr/bin/watchcat.sh "periodic_reboot" "$period" "$forcedelay"
 		;;
 	ping_reboot)
-		/usr/bin/watchcat.sh "ping_reboot" "$period" "$forcedelay" "$pinghosts" "$pingperiod" "$pingsize" &
-		logger -p user.info -t "watchcat" "started task (mode=$mode;period=$period;pinghosts=$pinghosts;pingperiod=$pingperiod;forcedelay=$forcedelay;pingsize=$pingsize)"
+		procd_set_param command /usr/bin/watchcat.sh "ping_reboot" "$period" "$forcedelay" "$pinghosts" "$pingperiod" "$pingsize"
 		;;
 	restart_iface)
-		/usr/bin/watchcat.sh "restart_iface" "$period" "$pinghosts" "$pingperiod" "$pingsize" "$interface" "$mmifacename" &
-		logger -p user.info -t "watchcat" "started task (mode=$mode;period=$period;pinghosts=$pinghosts;pingperiod=$pingperiod;pingsize=$pingsize;interface=$interface;mmifacename=$mmifacename;unlockbands=$unlockbands)"
+		procd_set_param command /usr/bin/watchcat.sh "restart_iface" "$period" "$pinghosts" "$pingperiod" "$pingsize" "$interface" "$mmifacename"
 		;;
 	*)
 		echo "Error starting Watchcat service. Invalid mode selection: $mode"
 		;;
 	esac
 
-	echo $! >>"${PIDFILE}.pids"
+	procd_set_param respawn ${respawn_threshold:-3600} ${respawn_timeout:-5} ${respawn_retry:-5}
+	procd_close_instance
 }
 
-stop() {
-	if [ -f "${PIDFILE}.pids" ]; then
-		logger -p user.info -t "watchcat" "stopping all tasks"
-
-		while read pid; do
-			kill -KILL "$pid"
-		done <"${PIDFILE}.pids"
-
-		rm "${PIDFILE}.pids"
-
-		logger -p user.info -t "watchcat" "all tasks stopped"
-	else
-		logger -p user.info -t "watchcat" "no tasks running"
-	fi
+start_service() {
+	config_load watchcat
+	config_foreach config_watchcat watchcat
+	procd_set_param file /etc/config/watchcat
```

Comment:
Does this work? I think it belongs into the foreach loop to the procd object.

Reply:
Yes, it works

```
 1864 root      1264 S    /bin/sh /usr/bin/watchcat.sh restart_iface 21600 8.8.8.8 30 standard wan
 1865 root      1264 S    /bin/sh /usr/bin/watchcat.sh periodic_reboot 21600 60
```